### PR TITLE
Add an API and CLI

### LINKED ISSUE #1: Add an API
Maybe we should be able to do that in nodejs scripts : 
```javascript
const findUnusedSassVar = require('find-unused-sass-variables')
const folders = []
const unUsed = findUnusedSassVar.find(folders)

### LINKED ISSUE #2: Add CLI
I have [a branch](https://github.com/XhmikosR/find-unused-sass-variables/tree/cli) with pretty basic CLI support. It works OK but I want to get rid of the global variable I'm using and then merge.

We could add a package like commander if needed, but preferably we should keep it as simple as possible since this tool is pretty dump.

Note to self: document caveats.

### PR DESCRIPTION
**Pull request by Johann-S**
Will allow us to have to separate executable one for the CLI (`cli.js`) and the second one for the api (`index.js`)

Fixes #1, fixes #2

### CODE REVIEW

**Comment by XhmikosR** [comment]
@Johann-S: thanks for this!

Now, only thing I'm thinking is why do we need an api.js file in the first place? I think we should have a cli file and an index file.

**Comment by Johann-S** [comment]
yep `index.js` and `cli.js` will be fine 👍

File: src/findUnusedVars.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// Blame TC39... https://github.com/benjamingr/RegExp.escape/issues/37
+function regExpQuote(str) {
+    return str.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');
+}
+
+function findUnusedVars (dir, isCLI) {
+    if (!(fs.existsSync(dir) && fs.statSync(dir).isDirectory())) {
+        if (isCLI) {
+            console.log(`"${dir}": Not a valid directory!`);
+            process.exit(1);
```
Comment (XhmikosR) [inline src/findUnusedVars.js]: Perhaps these checks should just be in the cli module? And just return something else here?

Comment (Johann-S) [reply src/findUnusedVars.js]: yep but I don't want to duplicate some of your code, that's why I added those checks

Comment (XhmikosR) [reply src/findUnusedVars.js]: How about just returning `-1` or something and the CLI will take care of exiting?

**Comment by XhmikosR** [comment]
@Johann-S: can you take care of the refactoring? After that I think we can merge :)

**Comment by Johann-S** [comment]
Do you want I rename api.js --> index.js and index.js --> cli.js ?

**Comment by XhmikosR** [comment]
I think that would be the best: 2 files, one for the lib and one for the cli. Thanks!

**Comment by Johann-S** [comment]
Done 👍

**Comment by XhmikosR** [comment]
Can't we just merge src/findUnusedVars.js with index.js?

**Comment by Johann-S** [comment]
sure that's can be done 👍

**Comment by XhmikosR** [comment]
I think we should follow "KISS" since it's a pretty simple tool :)

**Comment by XhmikosR** [comment]
Awesome, thanks! We are almost there :)

I just feel like some of the console.log we have in index.js don't belong there. Not sure if there's a right way to do it or not though.

**Comment by XhmikosR** [comment]
@Johann-S: I pushed a couple more patches. If you have any idea how to move the last console.log to the CLI feel free to push :)

**Comment by Johann-S** [comment]
I don't see any console.log in `index.js` so I think everything is fine 👍

**Pull request merged**